alfresco-search/src/main/java/org/alfresco/solr/tracker/MetadataTracker.java

```diff
@@ -591,7 +597,20 @@ protected Transactions getSomeTransactions(BoundedDeque<Transaction> txnsFound,
         Transactions transactions;
         // step forward in time until we find something or hit the time bound
         // max id unbounded
-        Long startTime = fromCommitTime == null ? Long.valueOf(0L) : fromCommitTime;
+        Long startTime = fromCommitTime == null  ? 0L : fromCommitTime;
+        log.debug(String.format("#### %s MetadataTracker getSomeTransactions start time: %d end: %d",
+                  this.coreName, 
+                  startTime,
+                  endTime));
+        if(startTime == 0)
+        {
+            return client.getTransactions(startTime,
+                                          null,
+                                          startTime + actualTimeStep, 
+                                          null, 
+                                          maxResults, 
+                                          shardstate);
+        }
```

Comment:
The while statement below will continue to try to get transactions when startTime > endTime if transactions are being returned. This seems wrong to me.  Instead I think the logic should be more like:
```
while ((startTime < endTime) && (size() == 0 || alreadyFound()))
```

i.e. We stop if we're past the end time, or if we haven't already seen some transactions.

Reply:
In fact the whole of the if statement and do-while loop can be simplified to:
```
do
{
    transactions = client.getTransactions(...);
    startTime += actualTimeStep;
    // Stop if we're looking for the first transactions, we're past the endTime or we've found some unseen transactions.
} while (startTime != actualTimeStep && startTime < endTime && (transactions.size() == 0 || alreadyFoundTransactions()));
return transactions;
```